app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -63,20 +63,25 @@ def kickstart_repos(host, content_facet: nil)
       def variant_repos(host, variant)
         if variant && host.content_source
           product_id = host.try(:content_facet).try(:kickstart_repository).try(:product_id) || host.try(:kickstart_repository).try(:product_id)
-          distros = distribution_repositories(host)
+          distribution_repositories(host)
             .joins(:product)
             .where("#{Katello::Repository.table_name}.distribution_variant LIKE :variant", { variant: "%#{variant}%" })
             .where("#{Katello::Product.table_name}.id": product_id).collect { |repo| repo.to_hash(host.content_source, true) }
-          distros
         end
       end
 
       def distribution_repositories(host, content_facet: nil)
         content_facet ||= host.content_facet
-        content_view = content_facet.try(:content_view) || host.try(:content_view)
-        lifecycle_environment = content_facet.try(:lifecycle_environment) || host.try(:lifecycle_environment)
-        if content_view && lifecycle_environment && host.os && host.architecture
-          Katello::Repository.in_environment(lifecycle_environment).in_content_views([content_view]).
+        case content_facet
+        when ::Katello::Host::ContentFacet
+          content_views = content_facet.try(:content_views) || host.try(:content_views)
```

Comment:
For some reason, for my host. `host.content_views` and `content_facet.content_views` returns [], but `single_content_view` returns what is correct.